conf.d/virtualfish.fish

```diff
@@ -1,4 +1,4 @@
-if set -q $VIRTUALFISH_PYTHON
+if not set -q $VIRTUALFISH_PYTHON; or set -q $VIRTUALFISH_PYTHON
```

Comment:
This is *also* wrong.

What happens here is that fish sees `set -q $VIRTUALFISH_PYTHON`, then expands the variable, and tries to see if those values name any variables.

That means after `set VIRTUALFISH_PYTHON python` it would *still* be false unless you have a *variable* called "python". But if you did e.g. `set VIRTUALFISH_PYTHON version` it would be true because $version exists.

What you want, and what you always wanted, is:

```fish
if set -q VIRTUALFISH_PYTHON
```

without the `$`.

This was always broken, it just happened to not fail in a way you noticed.

Reply:
Essentially, what happened before was that the if-branch was "almost always" taken. The `if` was basically pointless.

This proposed change makes it so the if-branch is again almost always taken. The real fix is to make the if work as intended.

Reply:
You are correct. I have corrected this. I have done more local testing to make sure this only runs the if when `VIRTUALFISH_PYTHON` is not set.

```suggestion
if not set -q VIRTUALFISH_PYTHON
```